src/dbnode/integration/large_tiles_test.go

```diff
@@ -111,62 +121,168 @@ func TestReadAggregateWrite(t *testing.T) {
 	processedBlockCount, err := testSetup.DB().AggregateTiles(storageOpts.ContextPool().Get(), srcNs.ID(), trgNs.ID(), aggOpts)
 	log.Info("Finished aggregation", zap.Duration("took", time.Since(start)))
 	require.NoError(t, err)
-	assert.Equal(t, int64(6), processedBlockCount)
+	assert.Equal(t, int64(10), processedBlockCount)
 
-	counters := reporter.Counters()
-	writeErrorsCount, _ := counters["database.writeAggData.errors"]
-	require.Equal(t, int64(0), writeErrorsCount)
-	seriesWritesCount, _ := counters["dbshard.large-tiles-writes"]
-	require.Equal(t, int64(2), seriesWritesCount)
+	require.True(t, xclock.WaitUntil(func() bool {
+		counters := reporter.Counters()
+		writeErrorsCount, _ := counters["database.writeAggData.errors"]
+		require.Equal(t, int64(0), writeErrorsCount)
+		seriesWritesCount, _ := counters["dbshard.large-tiles-writes"]
+		fmt.Println(seriesWritesCount)
+		return seriesWritesCount >= 2
+	}, time.Second*10))
 
 	log.Info("validating aggregated data")
-	expectedDps := map[int64]float64{
-		dpTimeStart.Add(110 * time.Minute).Unix(): 53.1,
-		dpTimeStart.Add(170 * time.Minute).Unix(): 59.1,
-		dpTimeStart.Add(230 * time.Minute).Unix(): 65.1,
-		dpTimeStart.Add(290 * time.Minute).Unix(): 71.1,
-		dpTimeStart.Add(350 * time.Minute).Unix(): 77.1,
+	expectedDps := []ts.Datapoint{
+		ts.Datapoint{Timestamp: dpTimeStart.Add(110 * time.Minute), Value: 53.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(170 * time.Minute), Value: 59.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(230 * time.Minute), Value: 65.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(290 * time.Minute), Value: 71.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(350 * time.Minute), Value: 77.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(410 * time.Minute), Value: 83.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(470 * time.Minute), Value: 89.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(530 * time.Minute), Value: 95.1},
+		ts.Datapoint{Timestamp: dpTimeStart.Add(590 * time.Minute), Value: 101.1},
 	}
+
 	fetchAndValidate(t, session, trgNs.ID(), ident.StringID("foo"), dpTimeStart, nowFn(), expectedDps)
 
-	expectedDps = map[int64]float64{
-		dpTimeStart.Unix(): 15,
+	expectedDps = []ts.Datapoint{
+		ts.Datapoint{Timestamp: dpTimeStart, Value: 15},
 	}
 	fetchAndValidate(t, session, trgNs.ID(), ident.StringID("aab"), dpTimeStart, nowFn(), expectedDps)
 }
 
+var (
+	iterationCount      = 100
+	testSeriesCount     = 100
+	testDataPointsCount = int(blockSizeT.Hours()) * 100
```

Comment:
nit: For consistency use "Datapoint" instead of "DataPoint" since that's what we do elsewhere, it seems that "datapoint" is considered an alternative spelling of "data point" in english dictionaries, hence no need to camel case the "Point".